smdebug/xgboost/utils.py

```diff
@@ -376,22 +405,38 @@ def parse_tree_model(booster, iteration, fmap=""):
                 missings.append(float("NAN"))
                 gains.append(float(stats[1]))
                 covers.append(float(stats[3]))
+                categories.append(float("NAN"))
             # Not a Leaf Node
             else:
                 # parse string
-                fid = arr[1].split("]")
-                parse = fid[0].split("<")
-                stats = re.split("=|,", fid[1])
+                fid = arr[1].split(']')
+                # these conditions brings backward compatible
+                if fid[0].find("<") != -1:
+                    # numerical
+                    parse = fid[0].split('<')
+                    splits.append(float(parse[1]))
+                    categories.append(float("NAN"))
+                elif fid[0].find(":{") != -1:
+                    # categorical
+                    parse = fid[0].split(":")
+                    cats = parse[1][1:-1]  # strip the {}
+                    cats_split = cats.split(",")
+                    splits.append(float("NAN"))
+                    print(f"debug {cats_split}")
```

Comment:
Should we remove this debug statement?

Reply:
yes, will remove in next commit, thanks for catching this!